### PR TITLE
Restructure for Typst Universe

### PR DESCRIPTION
**Pull request by mielpeeters**
I've updated the directory structure to comply with Typst's guidelines for templates and packages.
This way, we should be able to push this to the official Typst Universe!

Also, I updated some details such that the template is now compatible with the `v0.11.0` compiler.

### CODE REVIEW

File: 0.1.0/lib.typ [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+// workaround for lack of std scope
+#let std-biblio = bibliography
+
+// store in states such that these need to be passed
+// only once, in the `thesis` function
+#let _authors = state("authors", ())
+#let _title = state("title", [])
+#let _supervisors = state("supervisors", ())
+#let _year = state("year", [2023-2024])
```
Comment (CedricMeu) [inline 0.1.0/lib.typ]: Might be nice to have the default year be [the current year](https://typst.app/docs/reference/foundations/datetime/)?

File: 0.1.0/lib.typ [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+#let thesis(
```
Comment (CedricMeu) [inline 0.1.0/lib.typ:5]: Don't know if it's useful to have default values for the parameters here. These should all be overridden anyway (except for year, look at previous comment).

File: 0.1.0/template/images/ea.png
Comment (CedricMeu) [inline 0.1.0/template/images/ea.png:1]: I think we can remove the front-page (and therefore these images) for now. It messes up the page numbering.

Comment (CedricMeu) [reply 0.1.0/template/images/ea.png:1]: Unless you got that working?

File: 0.1.0/template/main.typ [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#import "@local/ugent-thesis:0.1.0": thesis, acronyms, front-page, page-content
+
+#show: thesis.with(
+  title: [A UGent Master's Dissertation Created Using Typst],
```
Comment (CedricMeu) [inline 0.1.0/template/main.typ:4]: Replace "Dissertation" with "Thesis" for consistency?

File: 0.1.0/lib.typ [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      ).find(h => h.location().page() == here().page())
+
+      let page_has_no_heading = headings_at_this_page == none 
+
+      if elems.len() != 0 and page_has_no_heading {
+        let body = elems.last().body
+        align(right, emph(body))
+      }
+    },
+    numbering: "I"
```
Comment (CedricMeu) [inline 0.1.0/lib.typ]: The numbering is set to roman numerals here, but never set back to normal. Or am i missing something?

File: 0.1.0/lib.typ [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  body
+
+  if bibliography != none {
+    pagebreak()
+    set std-biblio(style: "ieee", title: [References])
+    bibliography
+  }
+}
+
+#let page-content(body) = {
```
Comment (CedricMeu) [inline 0.1.0/lib.typ]: Should also reset page numbering here?

File: 0.1.0/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+## Dependencies
+
+- The library [acrostiche](https://typst.app/universe/package/acrostiche) is used to manage acronyms.
+    These are maintained under `src/acronyms.typ`.
+
+## Other remarks
+
+The `front-page` function is really only a placeholder for the eventual front page which you will have to export
+from Plato yourself.
```
Comment (CedricMeu) [inline 0.1.0/README.md]: Maybe just leave the front page out, might lead to someone not noticing it isn't the correct front page. Would be a painful mistake to make.

**Review by CedricMeu — CHANGES_REQUESTED** [review]
Thank you very much for this PR!

I have only a few small remarks but these should be easy to fix.

File: 0.1.0/lib.typ (hunk 2 above)
Comment (mielpeeters) [reply 0.1.0/lib.typ:5]: You're right, but to keep these arguments named instead of positional, we should maybe have `none` as default value.
Using this function without named arguments would become messy, I think.

**Comment by mielpeeters** [comment]
All comments were valid. I solved all issues which you addressed.

File: 0.1.0/test.typ [hunk 7]
```diff
@@ -0,0 +1,32 @@
+#import "lib.typ": thesis, acronyms, front-page, page-content, show-acronyms
```
Comment (CedricMeu) [inline 0.1.0/test.typ]: `front-page` doesn't exist anymore

File: 0.1.0/test.typ [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+#show: thesis.with(
+  title: [A UGent Master's Dissertation Created Using Typst],
+  authors: ("John Doe",),
+  supervisors: ("Prof. Dr. Jane Doe",),
+  year: "2023-2024",
+  bibliography: bibliography("template/bib.bib")
+)
+
+#front-page(
```
Comment (CedricMeu) [inline 0.1.0/test.typ]: Same here

File: 0.1.0/test.typ [hunk 9]
```diff
@@ -0,0 +1,32 @@
+#import "lib.typ": thesis, acronyms, front-page, page-content, show-acronyms
+
+#show: thesis.with(
+  title: [A UGent Master's Dissertation Created Using Typst],
+  authors: ("John Doe",),
+  supervisors: ("Prof. Dr. Jane Doe",),
+  year: "2023-2024",
+  bibliography: bibliography("template/bib.bib")
```
Comment (CedricMeu) [inline 0.1.0/test.typ]: Bibliography also isn't an argument anymore

File: 0.1.0/test.typ (hunk 7 above)
Comment (CedricMeu) [reply 0.1.0/test.typ]: The comments on this file also need to be fixed on `template/main.typ`

File: 0.1.0/lib.typ [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+      ).find(h => h.location().page() == here().page())
+
+      let page_has_no_heading = headings_at_this_page == none 
+
+      if elems.len() != 0 and page_has_no_heading {
+        let body = elems.last().body
+        align(right, emph(body))
+      }
+    },
+    numbering: "I"
```
Comment (CedricMeu) [inline 0.1.0/lib.typ]: For some reason the roman numbering on the first pages doesn't show op on my build of `test.typ`

**Review by CedricMeu — CHANGES_REQUESTED** [review]
Some of the functions you removed are still in `test.typ` and `main.typ`.  And a problem with numbering.

File: 0.1.0/lib.typ (hunk 10 above)
Comment (CedricMeu) [reply 0.1.0/lib.typ]: The page numbering (in roman) will start to show up starting on the 3rd page, as if it was the first page.

**Comment by mielpeeters** [comment]
I fixed the numbering problem!
I also reverted the acrostiche dependency removal because it should maybe be part of a next pull request

File: 0.1.0/README.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+Run the following command in your terminal. The font path needs to be supplied as an argument if 
+the UGent fonts are not installed system-wide. This command will create the output at `main.pdf`.
+
+```shell
+typst watch main.typ --font-path fonts
+```
+
+## Features
+- styling of headings and page numbering, etc.
+- `#todo[your todo note]`: a to-do note that is displayed as a colored block in the output.
```
Comment (CedricMeu) [inline 0.1.0/README.md]: Love the todo feature, but maybe put it in a different PR and diff against this one.

File: 0.1.0/lib.typ [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+    v(.75em)
+  }
+  
+  set text(font: font)
+  // don't break up words in justified text
+  set text(hyphenate: false)
+  
+  // make new sections appear on the right hand side
+  if odd_pagebreaks {
+    set pagebreak(weak: true, to: "odd")
```
Comment (CedricMeu) [inline 0.1.0/lib.typ:77]: how about we add an argument `pagebreak-to`, have it default `"odd"` and pass that to `to`. Or is that too much indirection?

File: 0.1.0/lib.typ [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  body
+}
+
+#let start-numbering(body) = {
+  set heading(numbering: "1.1", supplement: "Chapter")
+
+  show heading.where(level: 1): it => {
```
Comment (CedricMeu) [inline 0.1.0/lib.typ:86]: why is this so complex?

Comment (CedricMeu) [reply 0.1.0/lib.typ:86]: I mean, if it works alright but i can't wrap my head around it.

File: 0.1.0/lib.typ [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+#let page-content(body) = {
+  // from here, add a new page before each level one heading
+  show heading.where(
+    level: 1
+  ): it => {
+    pagebreak()
+    it
+  }
```
Comment (CedricMeu) [inline 0.1.0/lib.typ]: Maybe move this to `thesis`. I prefer that ToCs also get moved to the next odd page.

Comment (CedricMeu) [reply 0.1.0/lib.typ]: Then maybe just remove this function as it only calls `start-numbering`. And try to find a better name for that one. maybe `thesis-content` or `main-content`.

**Review by CedricMeu — CHANGES_REQUESTED** [review]
My last remarks. I am sure we can finish this PR very soon.

File: 0.1.0/lib.typ (hunk 12 above)
Comment (mielpeeters) [reply 0.1.0/lib.typ:77]: Will anyone ever set it to something else than "odd" or not setting it at all?
If there's just these two scenarios, a bool is more clear I think

File: 0.1.0/lib.typ (hunk 13 above)
Comment (mielpeeters) [reply 0.1.0/lib.typ:86]: This is for a custom styling of the first-level headings, with a large number and the underlined section name.
The context block is required for getting the counter at that location, and the grid aligns the number and section name next to each other.

File: 0.1.0/lib.typ (hunk 14 above)
Comment (mielpeeters) [reply 0.1.0/lib.typ]: good ideas

File: 0.1.0/lib.typ (hunk 12 above)
Comment (mielpeeters) [reply 0.1.0/lib.typ:77]: I kept it as a boolean for now, feel free to change later

File: 0.1.0/README.md (hunk 11 above)
Comment (mielpeeters) [reply 0.1.0/README.md]: ✅

File: 0.1.0/lib.typ (hunk 13 above)
Comment (CedricMeu) [reply 0.1.0/lib.typ:86]: Alright, perfect!

**Review by CedricMeu — APPROVED** [review]

**Comment by CedricMeu** [comment]
Nice job!
